### PR TITLE
feat(stdlib): Add `fromArray` to `Queue` module

### PR DESCRIPTION
**Pull request by spotandjake**
Adds `fromArray` to the `Queue` module, helping to bring it more inline with the other stdlib data structures.

### CODE REVIEW

**Review by jozanza — APPROVED** [review]

**Comment by spotandjake** [comment]
Done

File: stdlib/queue.gr [hunk 1]
```diff
@@ -164,6 +164,29 @@ provide let copy = queue => {
   { size, array: Array.copy(array), headIndex, tailIndex }
 }
 
+/**
+ * Creates a queue from an array.
+ *
+ * @param arr: The array to convert
+ *
```
Comment (phated) [inline stdlib/queue.gr:171]:
```suggestion
```

File: stdlib/queue.gr [hunk 2]
```diff
@@ -164,6 +164,29 @@ provide let copy = queue => {
...
+ * @returns A queue containing all values from the array
+ *
+ * @since v0.6.0
+ */
+provide let fromArray = arr => {
+  let size = Array.length(arr)
+  let contents = if (size == 0) {
+    Array.make(16, None)
+  } else {
+    let contents = Array.make(size * 2, None)
```
Comment (phated) [inline stdlib/queue.gr:181]: I thought we had `Array.init`, don't we?

**Comment by spotandjake** [comment]
Applied those suggestions

File: stdlib/queue.gr [hunk 3]
```diff
@@ -164,6 +164,31 @@ provide let copy = queue => {
...
+    Array.make(16, None)
+  } else {
+    let contents = Array.init(size * 2, i => {
+      if (i < size) {
+        Some(arr[i])
+      } else {
+        None
+      }
+    })
+    contents
```
Comment (phated) [inline stdlib/queue.gr:187]:
```suggestion
    Array.init(size * 2, i => {
      if (i < size) {
        Some(arr[i])
      } else {
        None
      }
    })
```

**Comment by phated** [comment]
@spotandjake please resolve the conflicts 🙏

**Comment by spotandjake** [comment]
> @spotandjake please resolve the conflicts 🙏

Done

**Review by phated — APPROVED** [review]

**Pull request merged**